company-id-gui/src-tauri/src/main.rs

```diff
@@ -0,0 +1,721 @@
+    serde_json::to_writer_pretty(file, &file_data).map_err(|e| Error::FileError(e.into()))?;
+
+    Ok(account)
+}
+
+async fn get_credential_deployment_info(
+    id_obj: IdentityObjectV1<IpPairing, ArCurve, AttributeKind>,
+    wallet_context: WalletContext<'_>,
+    acc_index: u8,
+) -> Result<
+    (
+        AccountCredentialMessage<IpPairing, ArCurve, AttributeKind>,
+        AccountKeys,
+    ),
+    Error,
+> {
+    let WalletContext {
+        wallet,
+        ip_context,
+        id_index,
+        ip_index,
+        ..
+    } = wallet_context;
+
+    let policy: Policy<ArCurve, AttributeKind> = Policy {
+        valid_to:   id_obj.get_attribute_list().valid_to,
+        created_at: id_obj.get_attribute_list().created_at,
+        policy_vec: BTreeMap::new(),
+        _phantom:   Default::default(),
+    };
+
+    let randomness = wallet
+        .get_blinding_randomness(ip_index, id_index)
+        .context("Failed to get blinding randomness.")?;
+    let prf_key = wallet
+        .get_prf_key(ip_index, id_index)
+        .context("Failed to get PRF key.")?;
+    let id_cred_sec_scalar = wallet
+        .get_id_cred_sec(ip_index, id_index)
+        .context("Failed to get idCredSec.")?;
+
+    let id_cred_sec = PedersenValue::new(id_cred_sec_scalar);
+    let id_cred = IdCredentials { id_cred_sec };
+    let cred_holder_info = CredentialHolderInfo::<ArCurve> { id_cred };
+    let aci = AccCredentialInfo {
+        cred_holder_info,
+        prf_key,
+    };
+    let id_use_data = IdObjectUseData { aci, randomness };
+
+    let acc_data = get_account_keys(&wallet, ip_index, id_index, acc_index)?;
+    let credential_context = CredentialContext {
+        wallet,
+        identity_provider_index: ip_index.into(),
+        identity_index: id_index,
+        credential_index: acc_index,
+    };
+
+    let message_expiry = TransactionTime::seconds_after(5 * 60);
+    let (cdi, _) = create_credential(
+        ip_context,
+        &id_obj,
+        &id_use_data,
+        acc_index,
+        policy,
+        &acc_data,
+        &credential_context,
+        &Either::Left(message_expiry),
+    )
+    .context("Could not generate the credential.")?;
+
+    let acc_keys = AccountKeys {
+        keys:      [(CredentialIndex::from(0), acc_data)].into(),
+        threshold: AccountThreshold::ONE,
+    };
+
+    let acc_cred_msg = AccountCredentialMessage {
+        message_expiry,
+        credential: AccountCredential::Normal { cdi },
+    };
+    Ok((acc_cred_msg, acc_keys))
+}
+
+fn get_account_keys(
+    wallet: &ConcordiumHdWallet,
+    ip_index: u32,
+    id_index: u32,
+    acc_index: u8,
+) -> Result<CredentialData, Error> {
+    let secret = wallet
+        .get_account_signing_key(ip_index, id_index, acc_index as u32)
+        .context("Failed to get account signing key.")?;
+    let mut keys = std::collections::BTreeMap::new();
+    let public = ed25519_dalek::PublicKey::from(&secret);
```

Comment:
```suggestion
    let public = (&secret).into();
```

And you can remove the dependency on ed25519_dalek.